sdk/nodejs/cmd/dynamic-provider/index.ts

```diff
@@ -109,32 +109,25 @@ async function diffRPC(call: any, callback: any): Promise<void> {
         const req: any = call.request;
         const resp = new provproto.DiffResponse();
 
-        // If the provider itself has changed, do not delegate to the dynamic provider. Instead, simply report that the
-        // resource requires replacement. This allows the new resource to be created using the new provider and the old
-        // resource to be deleted using the old provider.
         const olds = req.getOlds().toJavaScript();
         const news = req.getNews().toJavaScript();
-        if (olds[providerKey] !== news[providerKey]) {
-            resp.setReplacesList([ providerKey ]);
-        } else {
-            const provider = getProvider(olds);
-            if (provider.diff) {
-                const result: any = await provider.diff(req.getId(), olds, news);
-
-                if (result.changes === true) {
-                    resp.setChanges(provproto.DiffResponse.DiffChanges.DIFF_SOME);
-                } else if (result.changes === false) {
-                    resp.setChanges(provproto.DiffResponse.DiffChanges.DIFF_NONE);
-                } else {
-                    resp.setChanges(provproto.DiffResponse.DiffChanges.DIFF_UNKNOWN);
-                }
-
-                if (result.replaces && result.replaces.length !== 0) {
-                    resp.setReplacesList(result.replaces);
-                }
-                if (result.deleteBeforeReplace) {
-                    resp.setDeletebeforereplace(result.deleteBeforeReplace);
-                }
+        const provider = getProvider(news);
```

Comment:
There was a useful comment on the previous logic - can we add one describing the goals of the new logic?

Reply:
Done.